frontend/src/components/MissionOverview/OngoingMissionCard.tsx

```diff
@@ -27,13 +31,77 @@ export function OngoingMissionCard({ mission }: MissionProps) {
         navigate(path)
     }
     return (
-        <StyledMissionCard variant="default" style={{ boxShadow: tokens.elevation.raised }} onClick={routeChange}>
-            <Typography variant="h6">INSPECTION</Typography>
+        <StyledMissionCard variant="default" style={{ boxShadow: tokens.elevation.raised }}>
+            <Typography variant="h6" onClick={routeChange}>
+                INSPECTION
+            </Typography>
             <Typography>{mission.name}</Typography>
             <HorisontalContent>
                 <MissionStatusDisplay status={mission.missionStatus} />
-                <MissionProgressDisplay />
+                <MissionProgressDisplay tasks={mission.tasks} />
+                <MissionControlButtons mission={mission} />
             </HorisontalContent>
         </StyledMissionCard>
     )
 }
+
+function MissionControlButtons({ mission }: MissionProps) {
+    const [status, setStatus] = useState(mission.missionStatus)
+    const apiCaller = useApi()
+    enum ControlButton {
+        Pause,
+        Stop,
+        Resume,
+    }
+
+    const handleClick = (button: ControlButton) => {
+        switch (button) {
+            case ControlButton.Pause: {
+                apiCaller.pauseMission(mission.robot.id)
+                setStatus(MissionStatus.Paused)
+                break
+            }
+            case ControlButton.Resume: {
+                apiCaller.resumeMission(mission.robot.id)
+                setStatus(MissionStatus.Ongoing)
+                break
+            }
+            case ControlButton.Stop: {
+                apiCaller.stopMission(mission.robot.id)
+                setStatus(MissionStatus.Cancelled)
+                break
+            }
+        }
+    }
+
+    const renderControlIcon = (status: MissionStatus) => {
+        if (status == MissionStatus.Paused) {
+            return (
+                <>
+                    <Icon
+                        name="play_circle"
+                        style={{ color: tokens.colors.interactive.primary__resting.rgba }}
+                        size={32}
+                        onClick={() => handleClick(ControlButton.Resume)}
+                    />
+                    <Icon
+                        name="stop"
+                        style={{ color: tokens.colors.interactive.primary__resting.rgba }}
+                        size={32}
+                        onClick={() => handleClick(ControlButton.Stop)}
+                    />
+                </>
+            )
+        }
+        return (
```

Comment:
Should we not be able to stop the mission without pausing it first?

Reply:
I think it is natural to have a two step process here, since canceling a mission shouldn't be done on a whim. During the user test, no users had any problem finding out how to stop the mission